symfony/plugins/orangehrmMarketPlacePlugin/lib/dao/MarketplaceDao.php

```diff
@@ -51,15 +54,34 @@ public function getInstalationPendingAddonIds()
             $q = Doctrine_Query::create()
                 ->select('id')
                 ->from('Addon c')
-                ->where('c.status = ?', 'Requested');
+                ->where('c.status = ?', self::REQUESTED);
             $value = $q->execute(array(), Doctrine::HYDRATE_ARRAY);
             return $value;
         } catch (Exception $e) {
             $this->getLogger()->error("Exception in getValue:" . $e);
             throw new DaoException($e->getMessage(), $e->getCode(), $e);
         }
     }
-
+    /**
+     * @return array $paidAddonIds
+     * @throws DaoException
+     */
+    public function getPaidAddonIds()
```

Comment:
better to have a generic function 

getAddonByStatus( string $status) : array {
}

Reply:
If I am going to do a code refactoring for a generic function I may need to thoroughly look into the other places also where this code may get affected. If so I think should be done under a separate task.

Reply:
'Paid' is a status of the addon and it's not good to couple variable status into the function name .. if we need to get all 'request' status addon then you have to write getRequestAddonIds() methods. 
we should follow open for extending and close for modification ( open close principle) concept when writing method and classes..